pkg/ebpf/bpf_log.go

```diff
@@ -191,16 +191,15 @@ func (t *Tracee) processBPFLogs(ctx context.Context) {
 				logger.Errorw("Incrementing BPF logs count", "error", err)
 			}
 
-		case lost := <-t.lostBPFLogChannel:
-			// When terminating tracee-ebpf the lost channel receives multiple "0 lost events" events.
-			// This check prevents those 0 lost events messages to be written to stderr until the bug is fixed:
-			// https://github.com/aquasecurity/libbpfgo/issues/122
-			if lost > 0 {
-				if err := t.stats.LostBPFLogsCount.Increment(lost); err != nil {
-					logger.Errorw("Incrementing lost BPF logs count", "error", err)
-				}
-				logger.Warnw(fmt.Sprintf("Lost %d ebpf logs events", lost))
+		case lost, ok := <-t.lostBPFLogChannel:
+			if !ok { // channel closed
+				continue
```

Comment:
Just for my understanding - if we continue here since the channel is closed, will we ever reach the ctx.Done() case below?

Reply:
@yanivagman Yes, since `<-ctx.Done()` is another select case clause. But your question made me ponder whether we should continue (putting a warning log) or return; I said this because the channel is controlled by another party and when closed it is the end of its useful life. @josedonizetti when you pick this for review, please take this in consideration.

Reply:
@geyslan Although it will get to the ctx done, if the ctx is not done and the channel is closed it will trigger a never blocking loop, because the select will always match on the closed channel, and continue, consuming CPU. When would this chan be closed?  Why do we care about the closing for `lostBPFLogChannel` but not for `bpfLogsChannel`?

Reply:
> Although it will get to the ctx done, if the ctx is not done and the channel is closed it will trigger a never blocking loop, because the select will always match on the closed channel, and continue, consuming CPU.

Indeed.

> When would this chan be closed?

They're finally closed by libbpfgo PerfBuffer.Stop() which is called by PerfBuffer.Close(), which in its turn is called after tracee's `ctx.Done()`:

![Screenshot from 2023-09-05 08-06-00](https://github.com/aquasecurity/tracee/assets/3372117/50da2ac1-8c26-42ff-9507-48688423824b)

> Why do we care about the closing for lostBPFLogChannel but not for bpfLogsChannel?

lostBPFLogChannel is not buffered.

Reply:
@josedonizetti your review opened my eyes to the correct cause of the "lost 0 events" output. Although the problem is mentioned in different goroutines, the cause is only in `processLostEvents`. Please take a look at this new PR: https://github.com/aquasecurity/tracee/pull/3438